calcom/cal.com
# fix: A user joining from invite link of a team doesn't automatically become member of the org

## What does this PR do?

- Add membership of the organization of the team in case of team invite link Fixes #12813
- Fix derived username from email in case of team invite  - For a verified domain of org1.com and email of member@org1.com, the username was wrongly created as `member-org1 `when it should be `member`.
- Also refactored invitation tests for org to abstract out a few things plus added the tests for the above scenarios


## Type of change

## How should this be tested?
- Create a team(Engineering) in an org(Org1 from org1.com). Also, verify that organization.
- Copy Invite link for Engineering Team
- Open the link in incognito
- Signup using the link with email member@org1.com. 
- Verify that the username is `member` and not `member-org1` 
- Also, verify that the user shows up as an organization member settings/organizations/members

## Mandatory Tasks
